homeassistant/components/nice_go/coordinator.py

```diff
@@ -70,7 +73,17 @@ def __init__(self, hass: HomeAssistant) -> None:
         self.email = self.config_entry.data[CONF_EMAIL]
         self.password = self.config_entry.data[CONF_PASSWORD]
         self.api = NiceGOApi()
-        self.ws_connected = False
+        self._unsub_connected: Callable[[], None] | None = None
+        self._unsub_data: Callable[[], None] | None = None
+        self._unsub_connection_lost: Callable[[], None] | None = None
+        self.connected = False
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self._async_ha_stop)
```

Comment:
Let's register this in `__init__.py`. Please don't forget to wrap it in entry.async_on_unload

Reply:
done